vm/sdk/src/spawn.rs

```diff
@@ -0,0 +1,55 @@
+// Template address is read from context
+pub fn spawn(blob: Vec<u8>) {
+  let blob_u32 = bytes_to_u32_vec(&blob);
+  athena_vm::syscalls::spawn(blob_u32.as_ptr(), blob.len());
+}
+
+fn bytes_to_u32_vec<T: AsRef<[u8]>>(bytes: T) -> Vec<u32> {
+  let mut chunks = bytes.as_ref().chunks_exact(4);
+  let mut result = chunks
+    .by_ref()
+    .map(|chunk| {
+      let mut bytes = [0u8; 4];
+      bytes.copy_from_slice(chunk);
+      u32::from_le_bytes(bytes)
+    })
+    .collect::<Vec<u32>>();
+
+  let remainder = chunks.remainder();
+  if !remainder.is_empty() {
+    let mut bytes = [0u8; 4];
+    bytes[..remainder.len()].copy_from_slice(remainder);
```

Comment:
Just double checking. This always zero-pads to the right, which when combined with LE ends up with zero padding to the left (as in the below test). Is this the intended behavior?

Reply:
I'm not sure I understand. Do you refer to this test?
```
fn convert_not_a_multiple_of_4() {
    let result = bytes_to_u32_vec([1, 2, 3, 4, 5, 6, 7]);
    assert_eq!(result, vec![0x04030201, 0x00070605]);
  }
```

The second word is not left-padded. This is just what hex representation looks like (the lower bytes are on the right). In memory, the underlying data is continuous. When this vec is later written into the "VM memory", it works as expected:


|address | 0 | 1 | 2| 3 |
|---|---|---|---|---|
|0x0 | 0x01 | 0x02| 0x03|0x04|
|0x1 | 0x05 | 0x06| 0x07|0x00|

Reply:
yes, okay, makes sense now. I think the right way to interpret my comment is, "with LE, we want to zero-pad to the right, i.e., add zeroes to the MSB."